### DISCUSSION TITLE
`BeatmapSetOverlay` doesn't load data when logging in with the overlay open

### DISCUSSION THREAD

**Discussion by Stedoss** [ppy/osu#22012] [Q&A]
When clicking on the `Details...` dropdown option on a song in the carousel (not logged in), you get a screen that says `Sign in to view the beatmap info`. Logging in shows the overlay UI as expected, however since the `GetBeatmapSetRequest` API request failed and did not get resent upon login it shows an infinite loading spinner. The user then has to exit the overlay and open it again for the data to be shown. Ideally the request should be resent and the data be shown in the overlay.

### Reproduction steps:
1. Log out (or just be in a logged out state)
2. Open a `BeatmapSetOverlay`, eg: right click a song in the beatmap carousel and click `Details...`
3. The overlay should say `Sign in to view the beatmap info`
4. Sign in to an account
5. Data in the overlay will not load

### Video

https://user-images.githubusercontent.com/29103029/210284346-68a6e277-b693-4ba5-a963-aedb8491b270.mp4

Been sitting on this one for a while now, I thought I could do some funky stuff with [saving/caching the beatmap request](https://github.com/Stedoss/osu/commit/98379d42c20bd93a25912d66e995580395063e70), but it probably needs to be resolved in a better thought out way (some generic way for overlays to store requests so they can be resent? not sure).

**Comment by frenzibyte** [comment]
What you have in the first commit of the branch you linked would be what I would've done (in addition to not performing any request if not logged in), similar to how beatmap listing overlay handles it. Saving requests sounds like unnecessary overengineering to me.

**Reply by Stedoss** [reply]
Agree, doesn't need any more complexity than that. The issue doesn't seem too important so will clean everything up on that branch and PR it as low priority once I get the chance, cheers :)
